src/components/legend/style.js

```diff
@@ -357,8 +357,30 @@ module.exports = function style(s, gd, legend) {
             var d0 = d[0];
             var w = boundLineWidth(d0.mlw, marker.line, MAX_MARKER_LINE_WIDTH, CST_MARKER_LINE_WIDTH);
 
-            p.style('stroke-width', w + 'px')
-                .call(Color.fill, d0.mc || marker.color);
+            p.style('stroke-width', w + 'px');
+
+            var fillColor = d0.mc || marker.color;
+
+            var getPatternAttr = function(mp, dflt) {
+                if(mp && Array.isArray(mp)) {
+                    if(mp.length > 0) return mp[0];
+                    else return dflt;
+                }
+                return mp;
+            };
+            var markerPattern = marker.pattern;
+            var patternShape = markerPattern && getPatternAttr(markerPattern.shape, '');
+
+            if(patternShape) {
+                var patternBGColor = getPatternAttr(markerPattern.bgcolor, null);
+                var patternSize = getPatternAttr(markerPattern.size, 1);
```

Comment:
Can we limit the pattern size in the legend to no larger than the legend symbol itself, ie [12 px](https://github.com/plotly/plotly.js/blob/39e35ccc02eb5d76ea7bbdc88be61a7cb1943918/src/components/legend/style.js#L351)? Otherwise a large pattern can be obscured.

I guess then `var patternID = trace.uid;` will not suffice, the legend pattern will need its own ID.

Reply:
I set the maximum pattern size of legend items to 12 px in 6e9cb40e81b62ad3472367e42da6975034abbba0.

Actually I prefer to make legend item sizes configurable, but it would be out of scope of this PR.

Reply:
Agreed - more legend configurability has been on our wishlist for some time, but it should not be done in this PR. https://github.com/plotly/plotly.js/commit/6e9cb40e81b62ad3472367e42da6975034abbba0 looks great 🎉